R/pkg/R/DataFrame.R

```diff
@@ -945,6 +947,19 @@ setMethod("[", signature(x = "DataFrame", i = "missing"),
             select(x, j)
           })
 
+#' @rdname select
+setMethod("[", signature(x = "DataFrame", i = "Column"),
+          function(x, i, j, ...) {
```

Comment:
Also FYI the `Matrix` package which also uses S4 is a good example for things like this
https://github.com/rforge/matrix/blob/master/pkg/Matrix/R/Matrix.R#L457